File: src/main/java/upbrella/be/store/dto/response/CurrentUmbrellaStoreResponse.java
package upbrella.be.store.dto.response;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import upbrella.be.umbrella.entity.Umbrella;

@Getter
@Builder
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
public class CurrentUmbrellaStoreResponse {

    private long id;
    private String name;

    public static CurrentUmbrellaStoreResponse fromUmbrella(Umbrella umbrella) {

        return CurrentUmbrellaStoreResponse.builder()
                .id(umbrella.getStoreMeta().getId())
                .name(umbrella.getStoreMeta().getName())
                .build();
    }
}

File: src/main/kotlin/upbrella/be/store/dto/response/CurrentUmbrellaStoreResponse.kt
package upbrella.be.store.dto.response

import upbrella.be.umbrella.entity.Umbrella

data class CurrentUmbrellaStoreResponse(
    // TODO: 이거 id가 nullable 체크 필요.. (엔티티부터)
    val id: Long?,
    val name: String
) {
    companion object {
        fun fromUmbrella(umbrella: Umbrella): CurrentUmbrellaStoreResponse {
            return CurrentUmbrellaStoreResponse(
                id = umbrella.storeMeta.id,
                name = umbrella.storeMeta.name
            )
        }
    }
}
